### PR TITLE
Fixes a memory leak in the caching of raw C-string in a JSON object.

### PR DESCRIPTION
**Pull request by dhruvsinghal**
Refer to https://github.com/facebook/watchman/commit/b6599bda54f63203a54d012df9c689c43dcd932d#commitcomment-18461888

### CODE REVIEW

File: thirdparty/jansson/value.c [hunk 1]
```diff
@@ -697,6 +697,13 @@ const char *json_string_value(const json_t *json)
         // Safe to return the buffer itself
         return value->buf;
     }
+
+    // If its not null-terminated, use a cached version that is null-terminated
```
Comment (sid0) [inline thirdparty/jansson/value.c]: `it's`

Comment (dhruvsinghal) [inline thirdparty/jansson/value.c]: LOL Fixed.

**Comment by facebook-github-bot-7** [comment]
@dhruvsinghal updated the pull request.

**Comment by sid0** [comment]
Pushed as c092710854808431b478e97b767297b01538e395, thanks.

**Comment by wez** [comment]
Thanks!

**Pull request closed without merging**